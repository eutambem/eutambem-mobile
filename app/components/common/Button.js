import React from 'react';
import ExternalButton from 'react-native-button';
import PropTypes from 'prop-types';

import { DARK_PURPLE, GREY } from '../../styles';

const color = type => (type === 'cancel' ? GREY : DARK_PURPLE);

const Button = props => (
  <ExternalButton
    style={{ fontSize: 17, color: color(props.type), padding: 7 }}
    {...props}
  >
    {props.title}
  </ExternalButton>
);

Button.propTypes = {
  type: PropTypes.string,
  title: PropTypes.string.isRequired,
};

Button.defaultProps = {
  type: 'confirm',
};

export default Button;
