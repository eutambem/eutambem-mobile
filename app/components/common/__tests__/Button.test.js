import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import Button from '../Button';

const props = {
  title: 'Test Button',
  onPress: sinon.spy(),
};

describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
