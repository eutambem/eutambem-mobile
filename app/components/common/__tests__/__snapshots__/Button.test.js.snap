// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<Button
  onPress={[Function]}
  style={
    Object {
      "color": "#961687",
      "fontSize": 17,
      "padding": 7,
    }
  }
  title="Test Button"
  type="confirm"
>
  Test Button
</Button>
`;
